### PR TITLE
feat(rooms): create adapter interface

### PR DESCRIPTION
**Pull request by akoushke**

### CODE REVIEW

File: src/adapters/RoomsAdapter.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import RoomsAdapter from './RoomsAdapter';
+
+describe('People Adapter Interface', () => {
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.test.js]:
```suggestion
describe('Room Adapter Interface', () => {
```

File: src/adapters/RoomsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  // eslint-disable-next-line no-unused-vars
+  getRoom(ID) {
+    return throwError(new Error('getRoom(ID) must be defined in RoomsAdapter'));
+  }
+
+  /**
+   * Returns an observable that emits an array of previous activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<ActivityID>>}
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.js]:
```suggestion
   * @returns {Observable.<Array.<string>>}
```

File: src/adapters/RoomsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  // eslint-disable-next-line no-unused-vars
+  getPreviousRoomActivities(ID) {
+    return throwError(new Error('getPreviousRoomActivities(ID) must be defined in RoomsAdapter'));
+  }
+
+  /**
+   * Returns an observable that emits an array of current activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<ActivityID>>}
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.js]:
```suggestion
   * @returns {Observable.<Array.<string>>}
```

File: src/adapters/RoomsAdapter.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  beforeEach(() => {
+    roomsAdapter = new RoomsAdapter();
+  });
+
+  test('getRoom() returns an observable', () => {
+    expect(rxjs.isObservable(roomsAdapter.getRoom())).toBeTruthy();
+  });
+
+  test('getPerson() throws a proper error message', (done) => {
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.test.js]:
```suggestion
  test('getRoom() throws a proper error message', (done) => {
```

File: src/adapters/RoomsAdapter.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import {throwError} from 'rxjs';
+
+import WebexAdapter from './WebexAdapter';
+
+/**
+ * This is a base class that defines the interface that maps room data.
+ * Developers that want to extend `RoomsAdapter` must implement all of its methods,
+ * adhering to the exact parameters and structure of the returned objects.
+ */
+export default class RoomsAdapter extends WebexAdapter {
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.js]: Document the `Room` object just after the class, like we do on the other adapters.
(See [`Person` in `PeopleAdapter`](https://github.com/webex/components/blob/master/src/adapters/PeopleAdapter.js#L31) and [`Activity` in `ActivitiesAdapter`](https://github.com/webex/components/blob/master/src/adapters/ActivitiesAdapter.js#L12))

File: src/adapters/RoomsAdapter.test.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  beforeEach(() => {
+    roomsAdapter = new RoomsAdapter();
+  });
+
+  test('getRoom() returns an observable', () => {
+    expect(rxjs.isObservable(roomsAdapter.getRoom())).toBeTruthy();
+  });
+
+  test('getRooms() throws a proper error message', (done) => {
```
Comment (lalli-flores) [inline src/adapters/RoomsAdapter.test.js]:
```suggestion
  test('getRoom() throws a proper error message', (done) => {
```

File: src/adapters/RoomsAdapter.js (hunk 5 above)
Comment (akoushke) [reply src/adapters/RoomsAdapter.js]: I came up with something for now

Comment (lalli-flores) [reply src/adapters/RoomsAdapter.js]: Looks good

**Pull request merged**